### PR TITLE
Maintenance alerts rebase

### PR DESCRIPTION
**Pull request by TylerSheffels**
First pass at site wide alerts @dannyroberts

### CODE REVIEW

File: corehq/util/markup.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# coding: utf-8
+import re
+from django.utils.html import format_html, escape
+from django.utils.safestring import mark_safe
+
+
+url_re = re.compile(
+    r"""(?i)\b((?:[a-z][\w-]+:(?:/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’]))"""
```
Comment (dimagimon) [inline corehq/util/markup.py]: E501 line too long (215 > 115 characters)

File: corehq/util/markup.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+
+def mark_up_urls(text):
+    """
+    >>> mark_up_urls("Please see http://google.com for more info.")
+    u'Please see <a href="http://google.com">http://google.com</a> for more info.'
+    >>> mark_up_urls("http://commcarehq.org redirects to https://commcarehq.org.")
+    u'<a href="http://commcarehq.org">http://commcarehq.org</a> redirects to <a href="https://commcarehq.org">https://commcarehq.org</a>.'
```
Comment (dimagimon) [inline corehq/util/markup.py]: E501 line too long (138 > 115 characters)

**Comment by TylerSheffels** [comment]
In order to create alerts in this version we need to run from a shell

`from corehq.apps.hqwebapp.models import MaintenanceAlert
MaintenanceAlert(active=True, text="""all my text goes here""").save()`

File: corehq/apps/hqwebapp/templatetags/menu_tags.py [hunk 3]
```diff
@@ -144,3 +145,18 @@ def format_sidebar(context):
...
+def maintenance_alert():
+    try:
+        alert = (MaintenanceAlert.objects
+                 .filter(active=True)
+                 .order_by('modified'))[0]
+    except IndexError:
+        return ''
+    else:
+        return format_html(
+            '<div class="alert alert-error" style="text-align: center; margin-bottom: 0;">{}</div>',
```
Comment (TylerSheffels) [inline corehq/apps/hqwebapp/templatetags/menu_tags.py]: @amsagoff This is where Danny had commented

Comment (TylerSheffels) [inline corehq/apps/hqwebapp/templatetags/menu_tags.py]: He asked if you prefer red background or yellow background for the alerts

**Pull request merged**